cmd/soroban-rpc/lib/preflight/src/lib.rs

```diff
@@ -185,25 +191,233 @@ fn preflight_host_function_or_maybe_panic(
     host.set_ledger_info(ledger_info.into());
 
     // Run the preflight.
-    let results = host.invoke_functions(invoke_hf_op.functions.into())?;
+    let results = host.invoke_functions(invoke_hf_op.clone().functions.into())?;
     let auth_payloads = host.get_recorded_auth_payloads()?;
 
     // Recover, convert and return the storage footprint and other values to C.
     let (storage, budget, events) = host.try_finish().unwrap();
 
-    let fp = storage_footprint_to_ledger_footprint(&storage.footprint)?;
-    let fp_cstr = CString::new(fp.to_xdr_base64()?)?;
+    let diagnostic_events = host_events_to_diagnostic_events(&events)?;
+    let (transaction_data, min_fee) = compute_transaction_data_and_min_fee(
+        &invoke_hf_op,
+        &CSnapshotSource { handle },
+        &storage,
+        &budget,
+        &diagnostic_events,
+    )?;
+    let transaction_data_cstr = CString::new(transaction_data.to_xdr_base64()?)?;
     Ok(CPreflightResult {
         error: null_mut(),
         results: scvals_to_c(results)?,
-        footprint: fp_cstr.into_raw(),
+        transaction_data: transaction_data_cstr.into_raw(),
+        min_fee: min_fee,
         auth: recorded_auth_payloads_to_c(auth_payloads)?,
-        events: host_events_to_c(events)?,
+        events: diagnostic_events_to_c(diagnostic_events)?,
         cpu_instructions: budget.get_cpu_insns_count(),
         memory_bytes: budget.get_mem_bytes_count(),
     })
 }
 
+fn compute_transaction_data_and_min_fee(
+    invoke_hf_op: &InvokeHostFunctionOp,
+    snapshot_source: &CSnapshotSource,
+    storage: &Storage,
+    budget: &Budget,
+    events: &Vec<DiagnosticEvent>,
+) -> Result<(SorobanTransactionData, i64), Box<dyn error::Error>> {
+    let soroban_resources = calculate_soroban_resources(snapshot_source, storage, budget, events)?;
+    let fee_configuration = get_fee_configuration(snapshot_source)?;
+
+    let transaction_resources = TransactionResources {
+        instructions: soroban_resources.instructions,
+        read_entries: soroban_resources.footprint.read_only.as_vec().len() as u32,
+        write_entries: soroban_resources.footprint.read_write.as_vec().len() as u32,
+        read_bytes: soroban_resources.read_bytes,
+        write_bytes: soroban_resources.write_bytes,
+        metadata_size_bytes: soroban_resources.extended_meta_data_size_bytes,
+        // Note: we could get a better transaction size if the full transaction was passed down to libpreflight
+        transaction_size_bytes: estimate_max_transaction_size(
+            invoke_hf_op,
+            &soroban_resources.footprint,
+        )?,
+    };
+    let (min_fee, ref_fee) =
+        compute_transaction_resource_fee(&transaction_resources, &fee_configuration);
```

Comment:
@dmkozh I am interpreting `min_fee` as the fee required to access resources, on top of which the transaction submitted needs to add the _Stellar classic_ fee. Is that correct?

Reply:
@tsachiherman , @2opremio , based on this statement and [response on the js-soroban-client pr for handling inclusion fee](https://github.com/stellar/js-soroban-client/pull/77#discussion_r1190451805) , I've tentatively [added 'suggestedInclusionFee' to expected rpc simulation response in js-soroban-client ](https://github.com/stellar/js-soroban-client/pull/77/files#diff-718319220445fb0cf5b6fe5c0d0c88085f8f32ee0a90d41b496037b00836b8c3R124), if you agree, would need to be added here on `SimulateTransactionResponse`, let me know if this is right direction, thanks!

Reply:
I had already added `MinResourceFee` to the `SimulateTransactionResponse`, isn't that enough?

Reply:
yes, if `MinResourceFee` sums in the 'inclusion fee' suggested by @dmkozh also, I wasn't sure if that was the case as he was mentioning this 'inclusion fee' as something that should be added on top by 'they' which I understood to be remote clients after getting simulation response?

Reply:
`MinResourceFee` needs to be added on top of the classic Stellar fee (typically the minimum base fee of 100).

Reply:
ok, I'm trying to confirm this rule on handling fees to complete [js-soroban-client](https://github.com/stellar/js-soroban-client/pull/77) when it prepares a users tx with preflight results, @tsachiherman [you mentioned a different approach of taking max](https://github.com/stellar/js-soroban-client/pull/77#discussion_r1189248525) of classic fee(already set on the tx by user) or MinResourceFee. Which approach is correct for applying MinResourceFee on a user's tx, Max comparison or Summation? And this means there will be no other suggested inclusion fees provided by simulation response?

Reply:
received clarification, the rule is to use summation with existing tx classic fee:
>given that it's called MinResourceFee, it would imply that the user would need to pay a minimum of MinResourceFee + classic fees. so, if the user provided an amount that is anywhere less then (MinResourceFee + classic fees), then we should bump that up to this number.  ( since we know it won't work otherwise )